projects/angular/src/utils/drag-and-drop/if-dragged.ts

```diff
@@ -22,20 +22,25 @@ export class ClrIfDragged<T> implements OnDestroy {
     @Optional()
     @SkipSelf()
     private container: ViewContainerRef,
-    @Optional() private dragEventListener: DragEventListenerService<T>
+    @Optional() private dragEventListener: DragEventListenerService<T>,
+    private ngZone: NgZone
   ) {
     if (!this.dragEventListener || !this.container) {
       throw new Error('The *clrIfDragged directive can only be used inside of a clrDraggable directive.');
     }
 
     this.subscriptions.push(
       this.dragEventListener.dragStarted.subscribe(() => {
-        this.container.createEmbeddedView(this.template);
+        // Note: the `dragStarted` emits outside of the Angular zone, see `DragEventListenerService#broadcast`.
+        // This won't trigger another change detection if it's already being run in the zone.
+        this.ngZone.run(() => this.container.createEmbeddedView(this.template));
       })
     );
     this.subscriptions.push(
       this.dragEventListener.dragEnded.subscribe(() => {
-        this.container.clear();
+        // Note: the `dragEnded` emits outside of the Angular zone, see `DragEventListenerService#broadcast`.
+        // This won't trigger another change detection if it's already being run in the zone.
+        this.ngZone.run(() => this.container.createEmbeddedView(this.template));
```

Comment:
It doesn't seem great that the function inside the subscription is copy-pasted. Would you consider DRYing the code out?

Reply:
I don't think it's possible to decouple the `ngZone.run`, I could move it into the method but that would overcomplicate the code much more.